programs/drift_vaults/src/state/vault_depositor.rs

```diff
@@ -135,47 +179,14 @@ impl VaultDepositor {
         vault_protocol: &mut Option<RefMut<VaultProtocol>>,
         vault_equity: u64,
     ) -> Result<Option<u128>> {
-        vault.apply_rebase(vault_protocol, vault_equity)?;
-        let mut rebase_divisor = None;
-
-        if vault.shares_base != self.vault_shares_base {
-            validate!(
-                vault.shares_base > self.vault_shares_base,
-                ErrorCode::InvalidVaultRebase,
-                "Rebase expo out of bounds"
-            )?;
-
-            let expo_diff = (vault.shares_base - self.vault_shares_base).cast::<u32>()?;
-
-            let _rebase_divisor = 10_u128.pow(expo_diff);
-
-            msg!(
-                "rebasing vault depositor: base: {} -> {} ",
-                self.vault_shares_base,
-                vault.shares_base
-            );
-
-            self.vault_shares_base = vault.shares_base;
-
-            let old_vault_shares = self.unchecked_vault_shares();
-            let new_vault_shares = old_vault_shares.safe_div(_rebase_divisor)?;
-
-            msg!("rebasing vault depositor: shares -> {} ", new_vault_shares);
-
-            self.update_vault_shares(new_vault_shares, vault)?;
-
-            self.last_withdraw_request.rebase(_rebase_divisor)?;
-
-            rebase_divisor = Some(_rebase_divisor);
+        if let Some(rebase_divisor) =
+            VaultDepositorBase::apply_rebase(self, vault, vault_protocol, vault_equity)?
```

Comment:
you cant do self.apply_rebase because the fn have the same name?

Reply:
it calls the impl's apply_rebase because the `VaultDepositor` and `TokenizedVaultDepositors` have slightly different checks before the common rebase logic